features/karaf/openhab-core/src/main/feature/feature.xml

```diff
@@ -90,6 +91,11 @@
 		<bundle>mvn:org.openhab.core.bundles/org.openhab.core.config.discovery.addon.ip/${project.version}</bundle>
 	</feature>
 
+	<feature name="openhab-core-config-discovery-addon-sddp" version="${project.version}">
+		<feature>openhab-core-base</feature>
+		<bundle>mvn:org.openhab.core.bundles/org.openhab.core.config.discovery.sddp/${project.version}</bundle>
```

Comment:
I think you removed the wrong line. You should still have:
```
<feature>openhab-core-config-discovery-addon</feature>
```

Reply:
... but I will make another build and try again .. :)

Reply:
I tried it like you suggested. There are two issues..

1. In `G:\OH4_TEST\runtime\system\org\openhab\core\bundles` the `org.openhab.core.config.discovery.sddp` module IS installed but `org.openhab.core.config.discovery.addon.sddp` is NOT.

![image](https://github.com/openhab/openhab-core/assets/893994/8358518a-9940-4a08-8e8a-17480f7a4fb9)

2. Adding the `org.openhab.core.config.discovery.addon.sddp` manually to `G:\OH4_TEST\addons` does cause the module to load; but it still reports 'Not all dependencies satisfied, cannot activate'. I am guessing because the jar is in the `addons` folder, it can find the path to the `runtime\system\org\openhab\core\bundles` folder. ??

Reply:
> EDIT: Do I need to do something in `openhab-webui` to force the finder to be installed on start-up? I did already follow the coding in OH core to support installation of the finder (as for UPNP and MDNS etc.) .. but I wonder if you had hard-wired something in openhab-webui ??

No. When the configuration property is available, it is used by the AddonSuggestionService to trigger the install of the specific suggestion finder feature. There is nothing in the UI that does that.

Reply:
>  When the configuration property is available, it is used by the AddonSuggestionService to trigger the install

Just out of interest, do I have to do something specific to make that property available? I seem to recall that for new AddonFinder extension classes the default behaviour (i.e. when there is no entry in the config file) is enabled. But perhaps the sate is disabled by default? In either case, changing the parameter setting via the UI seems not to load/unload the finder either..

Reply:
No, there should not be a need to set it. And it should have an impact when changing it from the UI.
Note that there is a bug for the IP add-on finder, where the default is not set to true, although it does install by default. You see that it has the wrong default value in the UI.

Reply:
> Note that there is a bug for the IP add-on finder, where the default is not set to true,

1. Where is the default (not) set?
 
2. I have been logging the AddonSuggestionService and I can see that it is trying to load all of the finders (including this one) at startup. It does this before the SsdpDiscoveryService tries to load, and again after it has loaded. But in neither  case does SddpAddonFinder seem actually to have been loaded. => I wonder if there is an issue with AddonSuggestionService when it is trying to load a finder that has its own @reference dependency on another module in core? (The other finders do also have @reference dependencies, but these, as mentioned, are linked to the UPNP and MDNS communication modules in openhab-tp.

Reply:
> 1. Where is the default (not) set?

In `bundles/org.openhab.core/src/main/resources/OH-INF/config/addons.xml`

Reply:
> 2\. I have been logging the AddonSuggestionService and I can see that it is trying to load all of the finders (including this one) at startup. It does this before the SsdpDiscoveryService tries to load, and again after it has loaded. But in neither  case does SddpAddonFinder seem actually to have been loaded. => I wonder if there is an issue with AddonSuggestionService when it is trying to load a finder that has its own @reference dependency on another module in core? (The other finders do also have @reference dependencies, but these, as mentioned, are linked to the UPNP and MDNS communication modules in openhab-tp.

See my comments below. You are referencing to the wrong bundle in the feature.